core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -13007,4 +13009,97 @@ private String toSqlWithBloat(RelNode root, int bloat) {
+        builder.getTypeFactory().createSqlType(SqlTypeName.BIGINT);
+    Set<SqlKind> sqlKindList = new HashSet<>();
+    sqlKindList.add(SqlKind.NULLS_FIRST);
+    windowOrderCollation.add(new RexFieldCollation(builder.literal(23), sqlKindList));
+
+    final RexNode windowRexNode = builder.getRexBuilder().makeOver(rankRelDataType,
+        SqlStdOperatorTable.RANK, rexArgs, partitionRexKey,
+        ImmutableList.copyOf(windowOrderCollation),
+        RexWindowBounds.UNBOUNDED_PRECEDING, RexWindowBounds.UNBOUNDED_FOLLOWING, true,
+        true, false, false, false);
+
+    RexNode minusRexNode =
+        builder.call(SqlStdOperatorTable.MINUS, windowRexNode, builder.literal(1));
+    RexNode multiplicationRex =
+        builder.call(SqlStdOperatorTable.MULTIPLY, minusRexNode, builder.literal(5));
+
+    final RexNode windowRexNodeOfCount = builder.getRexBuilder().makeOver(rankRelDataType,
+        SqlStdOperatorTable.COUNT, rexArgs, partitionRexKey,
+        ImmutableList.of(), RexWindowBounds.UNBOUNDED_PRECEDING,
+        RexWindowBounds.UNBOUNDED_FOLLOWING, true, true, false,
+        false, false);
+
+    RexNode finalRex =
+        builder.call(SqlStdOperatorTable.DIVIDE_INTEGER, multiplicationRex, windowRexNodeOfCount);
+    final RelNode root = builder
+        .scan("EMP")
+        .project(builder.alias(finalRex, "quantile"))
+        .build();
+
+    final String expectedBiqQuery = "SELECT CAST(FLOOR(((RANK() OVER (ORDER BY 23)) - 1) * 5 "
+        + "/ (COUNT(*) OVER (ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING))) AS INT64)"
+        + " AS quantile\n"
+        + "FROM scott.EMP";
+    assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBiqQuery));
+  }
+
+  @Test public void testQuantileFunctionWithQualify() {
+    final RelBuilder builder = relBuilder();
+    List<RexNode> rexArgs = new ArrayList<>();
+    List<RexNode> partitionRexKey = new ArrayList<>();
+    List<RexFieldCollation> windowOrderCollation = new ArrayList<>();
+    final RelDataType rankRelDataType =
+        builder.getTypeFactory().createSqlType(SqlTypeName.BIGINT);
+    Set<SqlKind> sqlKindList = new HashSet<>();
+    sqlKindList.add(SqlKind.NULLS_FIRST);
+    windowOrderCollation.add(new RexFieldCollation(builder.literal(23), sqlKindList));
+
+    final RexNode windowRexNode = builder.getRexBuilder().makeOver(rankRelDataType,
+        SqlStdOperatorTable.RANK, rexArgs, partitionRexKey,
+        ImmutableList.copyOf(windowOrderCollation),
+        RexWindowBounds.UNBOUNDED_PRECEDING, RexWindowBounds.UNBOUNDED_FOLLOWING, true,
+        true, false, false, false);
+
+    RexNode minusRexNode =
+        builder.call(SqlStdOperatorTable.MINUS, windowRexNode, builder.literal(1));
+    RexNode multiplicationRex =
+        builder.call(SqlStdOperatorTable.MULTIPLY, minusRexNode, builder.literal(5));
+
+    final RexNode windowRexNodeOfCount = builder.getRexBuilder().makeOver(rankRelDataType,
```

Comment:
Please refactor and move the Rex creation code portion to a new method, as it is currently repeated in both test cases.

Reply:
Done